src/status_im/ui/screens/wallet/events.cljs

```diff
@@ -96,6 +98,50 @@
  (fn [{:keys [web3 obj success-event]}]
    (ethereum/estimate-gas-web3 web3 (clj->js obj) #(re-frame/dispatch [success-event %2]))))
 
+(defn- validate-token-name! [web3 {:keys [address symbol name]}]
+  (erc20/name web3 address #(when (and (seq %2) ;;NOTE(goranjovic): skipping check if field not set in contract
+                                       (not= name %2))
+                              (let [message (i18n/label :t/token-auto-validate-name-error
+                                                        {:symbol   symbol
+                                                         :expected name
+                                                         :actual   %2
+                                                         :address  address})]
+                                (log/warn message)
+                                (js/alert message)))))
```

Comment:
can we dispatch here `:alert` event?

Reply:
Why? This is an effect function, it's perfectly fine to have a side effect in it. Why pollute the environment with another dispatch?

Reply:
oh ok, right, it's effect, then fine, thanks

Reply:
it would make sense to at least call show-popup instead of raw js/alert